docs/_design/map-controller.md

```diff
@@ -0,0 +1,399 @@
+you to list a desired state for all your friends every time.
+This means you cannot edit or delete your friends.
+
+This approach was not chosen because:
+
+1. We have to send the entire list of matching friends as one big hook request.
+   This complicates the user's hook code because they probably need to loop over
+   each friend.
+   It's also inefficient for patterns like *"for every X (where there are a lot
+   of X's), create a Y"* since we have to sync every X if any one of them
+   changes, and we can't process any of them in parallel.
+1. It's tied in with the CompositeController API, and doing something similar
+   for other APIs like DecoratorController would require both duplicated and
+   different effort (see [Decorator Resources](#decorator-resources)).
+1. It either forces you to use the same selector to find friends as you use to
+   claim children, or it complicates the API with multiple selectors for
+   different resources, which becomes difficult to reason about.
+1. If we force the same selector to apply to both friends and children,
+   we also force you to explicitly specify a meaningful set of labels.
+   You can't use selector generation (`controller-uid: ###`) for cases when you
+   don't need orphaning and adoption; your friends won't match that selector.
+
+### Decorator Resources
+
+Add a new type of resource to DecoratorController called a decorator resource,
+which contains objects that inform the behavior of the decorator.
+This would allow controllers that look at non-owned resources as part of
+computing the desired state of their children.
+
+In particular, you could use DecoratorController to create attachments
+(extra children) on a parent object, while basing your desired state on
+information in another object (the decorator resource) that is not owned by that
+parent.
+
+This approach was not chosen because:
+
+1. It's unclear how we would "link" objects of the decorator resource to
+   particular parent objects being processed.
+   Would we apply the parent selector to find decorator objects?
+   Or apply a selector inside the decorator object to determine if it matches
+   the parent object?
+   Whatever we choose, it will likely be unintuitive and confusing for users.
+1. It's unclear what should happen if multiple decorator objects match a single
+   parent object.
+   We could send multiple decorator objects to the hook, but that just passes
+   the complexity on to the user.
+1. It's unclear whether decorator objects are expected to take part in ownership
+   of the objects created.
+   Depending on the use case, users might want attachments to be owned by just
+   the parent, just the decorator, or both.
+   This configuration adds to the cognitive overhead of using the API,
+   and there's no one default that's more intuitive than the others.
+
+## Example
+
+The example use case we'll consider in this doc is a controller called
```

Comment:
I think this has the potential to be integrated to stateful operators that we are working on. :)